samples/drivers/firmware/arm_scmi/src/shell_reset.c

```diff
@@ -0,0 +1,202 @@
+/*
+ * SCMI Reset domain management protocol Shell interface
+ *
+ * Copyright (c) 2024 EPAM Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/firmware/scmi/reset.h>
+#include <zephyr/drivers/reset.h>
+#include <zephyr/shell/shell.h>
+
+#if defined(CONFIG_DT_HAS_ARM_SCMI_RESET_ENABLED)
```

Comment:
Should rather depend on `RESET_ARM_SCMI` otherwise you'd just end up with a build error if built w/o `CONFIG_RESET=y`. Also, might be able to handle this using Kconfig/CMakeLists.txt.